File: helpers/b64helper.rb
#!/usr/bin/env ruby
require 'base64'
cmd = ARGV[0]
data = STDIN.read
if cmd == 'encode'
  print Base64.strict_encode64(data)
elsif cmd == 'decode'
  begin
    print Base64.strict_decode64(data)
  rescue
    STDERR.puts "Bad base64"
    exit 2
  end
else
  STDERR.puts "Usage: b64helper.rb encode|decode"
  exit 1
end

File: helpers/b64helper.pl
#!/usr/bin/env perl
use strict;
use warnings;
use MIME::Base64;
my $cmd = shift @ARGV;
my $in = do { local $/; <STDIN> };
if ($cmd eq 'encode') {
    print encode_base64($in, '');
} elsif ($cmd eq 'decode') {
    my $out = eval { decode_base64($in) };
    if ($@) { print STDERR "Bad base64\n"; exit 2 }
    print $out;
} else {
    die "Usage: b64helper.pl encode|decode\n";
}
